File: src/main/java/org/sanguineous/jscommand/velocity/command/ReloadCommand.java
package org.sanguineous.jscommand.velocity.command;

import com.velocitypowered.api.command.SimpleCommand;
import net.kyori.adventure.text.Component;
import org.sanguineous.jscommand.velocity.JsCommand;

import java.io.FileNotFoundException;

public class ReloadCommand implements SimpleCommand {
    private final JsCommand plugin;

    public ReloadCommand(JsCommand plugin) {
        this.plugin = plugin;
    }

    @Override
    public boolean hasPermission(Invocation invocation) {
        return invocation.source().hasPermission("jscommand.reload");
    }

    @Override
    public void execute(Invocation invocation) {
        String[] args = invocation.arguments();
        if (args.length == 1) {
            if (args[0].equalsIgnoreCase("reload")) {
                try {
                    plugin.loadCommands();
                    invocation.source().sendMessage(Component.text("Successfully reloaded commands"));
                } catch (FileNotFoundException e) {
                    e.printStackTrace();
                }
            }
            if (args[0].equalsIgnoreCase("resetplayerdata")) {
                plugin.getPlayerData().clear();
                invocation.source().sendMessage(Component.text("Successfully reset playerData"));
            }
        }
    }
}

File: src/main/kotlin/org/sanguineous/jscommand/command/ReloadCommand.kt
package org.sanguineous.jscommand.command

import net.md_5.bungee.api.CommandSender
import net.md_5.bungee.api.chat.ComponentBuilder
import net.md_5.bungee.api.plugin.Command
import org.sanguineous.jscommand.JsCommand

class ReloadCommand(private val plugin: JsCommand) : Command("jscommand", "jscommand.reload") {
    override fun execute(sender: CommandSender?, args: Array<String>?) {
        if (args?.get(0) == "reload") {
            plugin.loadCommands()
            sender!!.sendMessage(*ComponentBuilder("Successfully reloaded commands").create())
        }
    }
}